onnx/backend/test/case/node/dropout.py

```diff
@@ -37,8 +41,133 @@ def export_default():  # type: () -> None
+        )
+
+        r = np.float32(0.1)
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        y = dropout(x, r)
+        expect(node, inputs=[x, r], outputs=[y], name='test_dropout_default_ratio')
+
+    @staticmethod
+    def export_default_mask():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x'],
+            outputs=['y', 'z'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        y, z = dropout(x, return_mask=True)
+        expect(node, inputs=[x], outputs=[y, z], name='test_dropout_default_mask')
+
+    @staticmethod
+    def export_default_mask_ratio():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x', 'r'],
+            outputs=['y', 'z'],
+        )
+
+        r = np.float32(0.1)
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        y, z = dropout(x, r, return_mask=True)
+        expect(node, inputs=[x, r], outputs=[y, z], name='test_dropout_default_mask_ratio')
+
+    # Training tests.
+
+    @staticmethod
+    def export_training_default():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x', 'r', 't'],
+            outputs=['y'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        r = np.float32(0.5)
+        t = np.bool_(True)
+        y = dropout(x, r, training_mode=t)
+        expect(node, inputs=[x, r, t], outputs=[y], name='test_training_dropout_default')
+
+    @staticmethod
+    def export_training_default_ratio_mask():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x', 'r', 't'],
+            outputs=['y', 'z'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        r = np.float32(0.5)
+        t = np.bool_(True)
+        y, z = dropout(x, r, training_mode=t, return_mask=True)
+        expect(node, inputs=[x, r, t], outputs=[y, z], name='test_training_dropout_default_mask')
+
+    @staticmethod
+    def export_training():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x', 'r', 't'],
+            outputs=['y'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        r = np.float32(0.75)
+        t = np.bool_(True)
+        y = dropout(x, r, training_mode=t)
+        expect(node, inputs=[x, r, t], outputs=[y], name='test_training_dropout')
+
+    @staticmethod
+    def export_training_ratio_mask():  # type: () -> None
+        node = onnx.helper.make_node(
+            'Dropout',
+            inputs=['x', 'r', 't'],
+            outputs=['y', 'z'],
+        )
+
+        x = np.array([-1, 0, 1]).astype(np.float32)
+        r = np.float32(0.75)
+        t = np.bool_(True)
+        y, z = dropout(x, r, training_mode=t, return_mask=True)
+        expect(node, inputs=[x, r, t], outputs=[y, z], name='test_training_dropout_mask')
```

Comment:
Wouldn't this test fail in ORT's onnx_test, as mask is random.

Reply:
After setting the seed attribute of dropout to zero and also in the reference implementation, tests on ORT pass.

Reply:
I have updated ORT test infra to count for number of zeros and make sure they are around the same count as baseline.